Source/Core/Components/zPartialPrisonerComp.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using HarmonyLib;
+using HugsLib;
+using PrisonLabor.Core.AI.WorkGivers;
+using PrisonLabor.Core.Other;
+using PrisonLabor.Core.Trackers;
+using RimWorld;
+using Verse;
+using Verse.AI;
+using Verse.Noise;
+
+namespace PrisonLabor.Core.Components
+{
+    public partial class PrisonerComp
+    {
+        private bool isHungry = false;
+        private bool isStarving = false;
+
+        public bool IsHungry => isHungry;
+        public bool IsStarving => isStarving;
+
+        private void TickLongPrisoner()
+        {
+            this.isHungry = pawn.needs.food.CurLevelPercentage < 0.15f;
+            this.isStarving = pawn.needs.food.CurLevelPercentage <= 0.001f;
+        }
+
+        private void TickLongWarden()
+        {
+            if (pawn.CurJobDef.defName != "PrisonLabor_SupervisePrisonLabor") { return; }
+
+            var room = pawn.GetRoom();
+            if (room != null)
+            {
+                foreach (int id in Tracked.Prisoners[room.ID])
+                {
+                    var comp = (PrisonerComp)Tracked.pawnComps[id];
+                    var prisoner = comp.pawn;
+
+                    if (prisoner.IsFighting() && prisoner.CurrentBed() != null)
+                    {
+                        Job nJob = JobMaker.MakeJob(new JobDef
```

Comment:
why not in jobdefs?

Reply:
Again Because i had to make sure the chain work.